config/config_test.go

```diff
@@ -1583,6 +1583,587 @@ func TestMigrateConfigTCF2EnforcePurposeFlags(t *testing.T) {
+			want_initialize_caches_timeout_ms:          1000,
+			want_initialize_caches_query:               "old_initialize_caches_query",
+			want_initialize_caches_amp_query:           "old_initialize_caches_amp_query",
+			want_poll_for_updates_refresh_rate_seconds: 1000,
+			want_poll_for_updates_timeout_ms:           1000,
+			want_poll_for_updates_query:                "old_poll_for_updates_query",
+			want_poll_for_updates_amp_query:            "old_poll_for_updates_amp_query",
+		},
+		{
+			description: "New config set, old config not set",
+			config:      storedReqestsConfigs.new,
+
+			want_connection_dbname:                     "new_connection_dbname",
+			want_connection_host:                       "new_connection_host",
+			want_connection_port:                       2000,
+			want_connection_user:                       "new_connection_user",
+			want_connection_password:                   "new_connection_password",
+			want_fetcher_query:                         "new_fetcher_query",
+			want_fetcher_amp_query:                     "new_fetcher_amp_query",
+			want_initialize_caches_timeout_ms:          2000,
+			want_initialize_caches_query:               "new_initialize_caches_query",
+			want_initialize_caches_amp_query:           "new_initialize_caches_amp_query",
+			want_poll_for_updates_refresh_rate_seconds: 2000,
+			want_poll_for_updates_timeout_ms:           2000,
+			want_poll_for_updates_query:                "new_poll_for_updates_query",
+			want_poll_for_updates_amp_query:            "new_poll_for_updates_amp_query",
+		},
+		{
+			description: "New config and old config set",
+			config:      storedReqestsConfigs.both,
+
+			want_connection_dbname:                     "new_connection_dbname",
+			want_connection_host:                       "new_connection_host",
+			want_connection_port:                       2000,
+			want_connection_user:                       "new_connection_user",
+			want_connection_password:                   "new_connection_password",
+			want_fetcher_query:                         "new_fetcher_query",
+			want_fetcher_amp_query:                     "new_fetcher_amp_query",
+			want_initialize_caches_timeout_ms:          2000,
+			want_initialize_caches_query:               "new_initialize_caches_query",
+			want_initialize_caches_amp_query:           "new_initialize_caches_amp_query",
+			want_poll_for_updates_refresh_rate_seconds: 2000,
+			want_poll_for_updates_timeout_ms:           2000,
+			want_poll_for_updates_query:                "new_poll_for_updates_query",
+			want_poll_for_updates_amp_query:            "new_poll_for_updates_amp_query",
+		},
+	}
+
+	for _, tt := range storedReqestsTests {
+		v := viper.New()
+		v.SetConfigType("yaml")
+		v.ReadConfig(bytes.NewBuffer(tt.config))
+
+		migrateConfigDatabaseConnection(v)
+
+		if len(tt.config) > 0 {
+			assert.Equal(t, tt.want_connection_dbname, v.Get("stored_requests.database.connection.dbname").(string), tt.description)
```

Comment:
Nitpick: instead of using type assertions you can call the viper methods `v.GetString` and `v.GetInt` instead of `v.Get` with the type assertion.

Reply:
Fixed